doc/src/sphinx/develop/extractors.rst

```diff
@@ -27,4 +34,268 @@ The current list of supported events is:
+
+
+Below are examples of each file:
+
+* `docker-compose.yml <https://github.com/clowder-framework/clowder/blob/develop/docker-compose.yml>`_
+    * This file sets up Clowder and its dependencies such as MongoDB and RabbitMQ. You should not have to modify it.
+* `docker-compose.override.yml <https://github.com/clowder-framework/clowder/blob/develop/docker-compose.override.example.yml>`_
+    * This file overrides defaults, and can be used to customize clowder. When downloading the file, make sure to rename it to docker-compose.override.yml. In this case it will expose clowder, mongo and rabbitmq ports to the localhost.
+* `docker-compose.extractor.yml <https://github.com/clowder-framework/clowder/blob/develop/docker-compose.extractors.yml>`_
+    * This file deploys your extractor to Clowder. You will have to update this file to reflect your extractor's name, Docker image name and version tag, and any other requirements like environment variables. See below:
+
+
+.. code-block:: yaml
+
+    version: '3.5'
+
+    services:
+      myextractor:
+        image: myextractor_imagename:mytag
+        restart: unless-stopped
+        networks:
+          - clowder
+        depends_on:
+          - rabbitmq
+          - clowder
+        environment:
+          - RABBITMQ_URI=${RABBITMQ_URI:-amqp://guest:guest@rabbitmq/%2F}
+          # Add any additional environment variables your code may need here
+      # Add multiple extractors below following template above
+
+
+1. Initialize Clowder. All the commands below assume that you are running this in a folder called tests, hence the network name tests_clowder. If you ran the docker-compose command in a folder called clowder, the network would be clowder_clowder.
+
+.. code-block:: bash
+
+	docker run -ti --rm --network tests_clowder clowder/mongo-init
+
+
+4. Enter email, first name, last name password, and admin: true when prompted.
+
+5. Navigate to localhost:9000 and login with credentials you created in step 4.
+
+6. Create a test space and dataset. Then click 'Select Files' and upload (if the file stays in CREATED and never moves to PROCESSED you might need to change the permission on the data folder using docker run -ti --rm --network tests_clowder clowder/mongo-init).
+
+7. Click on file and type submit for extraction.
+
+8. It may take a few minutes for you to be able to see the extractors available within Clowder.
+
+9. Eventually you should see your extractor in the list and click submit.
+
+10. Navigate back to file and click on metadata.
+
+11. You should see your metadata present if all worked successfully.
+
+
+A Quick Note on Debugging
+##########################
+
+To check the status of your extraction, navigate to the file within Clowder and click on the “Extractions” tab. This will give you a list of extractions that have been submitted. Any error messages will show up here if your extractor did not run successfully. 
+
+.. container:: imagepadding
```

Comment:
I had to remove the `.. container:: imagepadding` to get the images to show up.

Also the links all point to the same file `ug_extractors-1.png`.

I can make the changes but I wanted to check first that I wasn't missing 
something with the `.. container:: imagepadding`. Can I remove it? 

I am running sphinx-build 3.5.4

Thank you!

Reply:
You can remove the imagepadding. I wasn't doing anything special with it. As for the images that have the prefix ug_extractors-, the order for the images as they appear in the doc is 1,2,4,3. Would you be able to make these changes alongside removing the imagepadding?

Reply:
Thank you. Yes I can make the changes.